### PR TITLE
Create custom route my locations

### PR DESCRIPTION
**Pull request by chrononaut76**
1. Created custom route
2. Created controller action
3. Created view

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -39,8 +39,11 @@
             <%= image_tag "signup_avatar.jpeg", class: "avatar dropdown-toggle", id: "navbarDropdown", data: { bs_toggle: "dropdown" }, 'aria-haspopup': true, 'aria-expanded': false %>
             <div class="dropdown-menu dropdown-menu-end" aria-labelledby="navbarDropdown">
         <% if user_signed_in? %>
-              <%= link_to "My bookings", bookings_path, class:"dropdown-item" %>
-              <%= link_to "Log out",  destroy_user_session_path, class: "dropdown-item", data: {turbo_method: :delete} %>
```
Comment (lauradumi) [inline app/views/layouts/application.html.erb:51]: Here there will be conflicts with the pr I just merged. 
The "my bookings, and "add new location" are now in the nav bar as links and no longer in the drop down menu. 
So we would need to add the my locations link as a list to the nav-items.

File: app/controllers/locations_controller.rb [hunk 2]
```diff
@@ -1,31 +1,43 @@
 class LocationsController < ApplicationController
+  before_action :authenticate_user!, except: [:index, :show]  # Add this line to enforce authentication
```
Comment (sofofos) [inline app/controllers/locations_controller.rb:2]: 🙌

File: app/controllers/locations_controller.rb [hunk 3]
```diff
@@ -1,31 +1,43 @@
...
   end
 
   # GET /locations/new
   def new
-    @location = Location.new
+    if current_user.is_host
+      @location = Location.new
+    else
+      redirect_to root_path, status: :unauthorized
+    end
```
Comment (sofofos) [inline app/controllers/locations_controller.rb]: Similar to how we are using `set_location` on line 3, this logic could be exported to a private method, and validated with a `before_action` callback :) 
something like this: 
```
def validate_host
  redirect_to root_path, status: :unauthorized unless current_user.is_host
end
```
and the callback: 
```
before_action :validate_host, only: [:new, :edit, :destroy]
```
this way, your `new`, `edit`, and `destroy` methods can simply contain the code necessary for the host user's actions 🚀  
```suggestion
      @location = Location.new
```

**Review by sofofos — COMMENTED** [review]
Looking good! Just a couple of suggestions

File: app/views/layouts/application.html.erb (hunk 1 above)
Comment (chrononaut76) [reply app/views/layouts/application.html.erb:51]: Thanks for the feedback @lauradumi! I can comment these lines out so they don't create any merge conflicts. We can later reincorporate this code in a different PR.

File: app/controllers/locations_controller.rb (hunk 3 above)
Comment (chrononaut76) [reply app/controllers/locations_controller.rb]: Thanks for the recommendation @sofofos! I've incorporated these changes to my code.

**Review by lauradumi — APPROVED** [review]

**Pull request merged**